internal/repo/datasource/sql_test.go

```diff
@@ -0,0 +1,321 @@
+package datasource
+
+import (
+	"errors"
+	"github.com/DATA-DOG/go-sqlmock"
+	_ "github.com/go-sql-driver/mysql"
+	"github.com/vatsal278/TransactionManagementService/internal/config"
+	"github.com/vatsal278/TransactionManagementService/internal/model"
+	"reflect"
+	"regexp"
+	"strings"
+	"testing"
+	"time"
+)
+
+func TestSqlDs_HealthCheck(t *testing.T) {
+	db, _, err := sqlmock.New()
+	if err != nil {
+		t.Fail()
+	}
+	svcConfig := config.SvcConfig{
+		DbSvc: config.DbSvc{Db: db},
+	}
+	dB := NewSql(svcConfig.DbSvc, "newTemp")
+
+	tests := []struct {
+		name        string
+		setupFunc   func()
+		cleanupFunc func()
+		filter      map[string]interface{}
+		validator   func(bool)
+		dbInterface DataSourceI
+	}{
+		{
+			name: "SUCCESS::Health check",
+			validator: func(res bool) {
+				if res != true {
+					t.Errorf("Want: %v, Got: %v", true, res)
+				}
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+
+			res := dB.HealthCheck()
+
+			if tt.validator != nil {
+				tt.validator(res)
+			}
+		})
+	}
+}
+func TestSqlDs_Get(t *testing.T) {
+	tests := []struct {
+		name        string
+		setupFunc   func() sqlDs
+		cleanupFunc func()
+		filter      map[string]interface{}
+		validator   func([]model.Transaction, int, error)
+		dbInterface DataSourceI
```

Comment:
Don't see this field being used. Can this be removed?

Reply:
done